docs/sources/k6/next/using-k6-browser/how-to-write-browser-tests/asynchronous-operations.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Asynchronous operations
+description: 'Learn how the k6 browser module uses asynchronous operations.'
+weight: 100
+---
+
+# Asynchronous operations
+
+Since many browser operations happen asynchronously, and to follow the Playwright API more closely, we are working on migrating most of the browser module methods to be asynchronous as well.
+
+At the moment, methods such as `page.goto()`, `page.waitForNavigation()` and `Element.click()` return [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises), and scripts must be written to handle this properly.
+
+To avoid timing errors or other race conditions in your script, if you have actions that load up a different page, you need to make sure that you wait for that action to finish before continuing.
```

Comment:
I believe we're mostly done with this move, but I might be wrong, might be worth checking with the Browser folks 😇